api/backend/blueflood/blueflood.go

```diff
@@ -283,6 +346,12 @@ var samplerMap map[api.SampleMethod]sampler = map[api.SampleMethod]sampler{
 		bucketSampler: func(bucket []float64) float64 {
 			value := bucket[0]
 			for _, v := range bucket {
+				if math.IsNaN(v) {
+					continue
+				}
+				if math.IsNaN(value) {
+					value = v
+				}
 				value = math.Min(value, v)
```

Comment:
Should probably put this line an an else. It ends up being a noop when `IsNaN(value)`